src/app/update-modal/update-modal.component.ts

```diff
@@ -139,8 +144,12 @@ export class UpdateModalComponent implements OnInit, OnChanges {
     let time; 
     if (this.form.get('type').value == '2') {   // If Type is TIMED == 2
       time = Utils.convertTimeToSec(this.form.get('time').value);
-      let index = this.form.get('day').value != undefined ? this.days.indexOf(this.form.get('day').value) : 0;
-      this.form.controls['day'].setValue(index + 1);
+      let index = this.form.get('day').value != undefined ? (this.days.indexOf(this.form.get('day').value)+1) : 0;
```

Comment:
why +1 now?

Reply:
`indexOf()` returns index, which is starting from 0 but in our `weekDays` enum, index starts from 1. To match the index values, it required to use `+1` here.